config.yaml

```
@@ -0,0 +1,23 @@
+# include version specification in configuration files to help with parsing and schema evolution.
+version: 0.0.1
+sdk:
+  # Disable the SDK for all signals.
+  #
+  # Boolean value. If "true", a no-op SDK implementation will be used for all telemetry
+  # signals. Any other value or absence of the variable will have no effect and the SDK
+  # will remain enabled. This setting has no effect on propagators configured through
+  # the OTEL_PROPAGATORS variable.
+  #
+  # Environment variable: OTEL_SDK_DISABLED
+  disabled: false
+  # Configure resource attributes and resource detection for all signals.
+  resource:
+    # Key-value pairs to be used as resource attributes.
+    #
+    # Environment variable: OTEL_RESOURCE_ATTRIBUTES
+    attributes:
+      service:
+        # Sets the value of the `service.name` resource attribute
+        #
+        # Environment variable: OTEL_SERVICE_NAME
+        name: "unknown_service"
```

```suggestion
      # Sets the value of the `service.name` resource attribute
      #
      # Environment variable: OTEL_SERVICE_NAME
      service:\name: "unknown_service"
```
`attributes` as a flat map of key values for now.

Should consider whether the value's type is unambiguous. For example, if the attribute is `key: 10` does `10` map to a string, integer, or float? Alternatively could do something like:
```
attributes:
  -  key: service.name
      string_value: "unknown_service"
  - key: foo
     int_value: 10
```
More verbose yes, but also unambiguous which would address issues like [this](https://github.com/open-telemetry/opentelemetry-specification/issues/2857).

---

So yaml supports `!!int` to force a type, any thoughts on using that instead of having different field identifier for the different types? 

I guess having different field identifiers would allow us to support lists/arrays more easily in the future.

---

> So yaml supports !!int to force a type

Didn't realize that!

Suppose that provides an answer for list / arrays as well (i.e. `[!!int 10, !!int 15]`) but not super pretty. This alleviates most of my concern. Happy enough going with a `service.name: value` for now. Either way, I'm sure this won't be the last time it comes up 😁